Dr-Nekoma/RacketowerDB
# Constraints Serialization - A Petition against Mister 10

# Goal

We modeled how to add constraints to our database. And today, after a very engaging and fun session we managed to conquer its serialization, which involves writing Racket directly to the disk. The problem, once anticipated a few times already, is that depending on new lines as the spacers would eventually cause harm. That day has come. Scars and injuries were created because of the infamous ASCII equivalent of the new line, a.k.a, Mister 10. We should fight back with a proper serialization method, ideally with some documentation alongside it.

# How to Test

- Run `nix develop && just build && ./main` and you **should not** see any errors.

# Observations for the Reviewer

Someone needs to fix the serialization properly to be orthogonal to the amount of `\n` that we may have in the serialized `NDF` files.
